ietf/doc/views_draft.py

```diff
@@ -1857,3 +1883,122 @@ def set_intended_status_level(request, doc, new_level, old_level, comment):
+
+class IssueWorkingGroupLastCallForm(forms.Form):
+    end_date = DatepickerDateField(
+        required=True,
+        date_format="yyyy-mm-dd",
+        picker_settings={
+            "autoclose": "1",
+        },
+        help_text="The date the Last Call closes. If you change this, you must MANUALLY change the date in the subject and body below.",
+    )
+
+    to = MultiEmailField(
+        required=True,
+        help_text="Comma separated list of address to use in the To: header",
+    )
+    cc = MultiEmailField(
+        required=False, help_text="Comma separated list of addresses to copy"
+    )
+    subject = forms.CharField(
+        required=True,
+        help_text="Subject for Last Call message. If you change the date here, be sure to make a matching change in the body.",
+    )
+    body = forms.CharField(
+        widget=forms.Textarea, required=True, help_text="Body for Last Call message"
+    )
+
+    def clean_end_date(self):
+        end_date = self.cleaned_data["end_date"]
+        if end_date <= date_today(DEADLINE_TZINFO):
+            raise forms.ValidationError("End date must be later than today")
+        return end_date
+
+    def clean(self):
+        cleaned_data = super().clean()
+        end_date = cleaned_data.get("end_date")
+        if end_date is not None:
+            body = cleaned_data.get("body")
+            subject = cleaned_data.get("subject")
+            if end_date.isoformat() not in body:
+                self.add_error(
+                    "body",
+                    forms.ValidationError(
+                        f"Last call end date ({end_date.isoformat()}) not found in body"
+                    ),
+                )
+            if end_date.isoformat() not in subject:
+                self.add_error(
+                    "subject",
+                    forms.ValidationError(
+                        f"Last call end date ({end_date.isoformat()}) not found in subject"
+                    ),
+                )
+        return cleaned_data
+
+
+@login_required
+def issue_wg_lc(request, name):
+    doc = get_object_or_404(Document, name=name)
+
+    if doc.stream_id != "ietf":
+        raise Http404
+    if doc.group is None or doc.group.type_id != "wg":
+        raise Http404
+
+    if not is_authorized_in_doc_stream(request.user, doc):
+        permission_denied(request, "You don't have permission to access this page.")
+
+    if request.method == "POST":
+        form = IssueWorkingGroupLastCallForm(request.POST)
+        if form.is_valid():
+            # Intentionally not changing tags or adding a comment
+            # those things can be done with other workflows
+            by = request.user.person
+            prev_state = doc.get_state("draft-stream-ietf")
+            events = []
+            wglc_state = State.objects.get(type="draft-stream-ietf", slug="wg-lc")
+            doc.set_state(wglc_state)
+            e = add_state_change_event(doc, by, prev_state, wglc_state)
```

Comment:
I think this point can now be reached with `prev_state == wglc_state`, in which case `e` will be `None`. Is that intended (or is something preventing it)?

Reply:
good catch - not allowing tags/comment in this form makes allowing the view questionable at all if the document is already currently in the wg-lc state. I'll work through whether being able to use this view to update WGLC in progress is the right thing to or not.